### PR TITLE
Bump controller-gen to v0.4.0 to get apiext/v1 generated CRDs

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What this PR does / why we need it**:
After merging https://github.com/nmstate/kubernetes-nmstate/pull/608 CRDs are back to
apiextensions.k8s.io/v1beta1, we have to upgrade controller-tools so they get generated
with apiextensions.k8s.io/v1.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RamLavi** [comment]
can you split the auto generated stuff to a different commit?

**Comment by qinqon** [comment]
@RamLavi done

File: api/shared/state_types.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ type RawState []byte
 //
 // [1] https://github.com/nmstate/nmstate/blob/master/libnmstate/schemas/operational-state.yaml
 // +k8s:openapi-gen=true
```
Comment (phoracek) [inline api/shared/state_types.go:11]: You dropped openapi targets in this PR. Does it make sense to keep these comments?

**Review by phoracek — CHANGES_REQUESTED** [review]

File: api/shared/state_types.go (hunk 1 above)
Comment (qinqon) [reply api/shared/state_types.go:11]: Good catch, removing.

**Review by phoracek — APPROVED** [review]
/lgtm
/approve

**Pull request merged**